control_panel_api/services.py

```diff
@@ -0,0 +1,135 @@
+import re
+
+from django.conf import settings
+from django.template.defaultfilters import slugify
+
+from . import aws
+
+READ_WRITE = 'readwrite'
+READ_ONLY = 'readonly'
+
+
+def _bucket_name(name):
+    """Prefix the bucket name with environment e.g. dev-james"""
+    return "{}-{}".format(settings.ENV, name)
+
+
+def _policy_name(bucket_name, readwrite=False):
+    """Prefix the policy name with bucket name, postfix with access level e.g. dev-james-readwrite"""
+    return "{}-{}".format(bucket_name, READ_WRITE if readwrite else READ_ONLY)
+
+
+def _policy_arn(bucket_name, readwrite=False):
+    """Return full bucket policy arn e.g. arn:aws:iam::1337:policy/bucketname-readonly"""
+    return "{}:policy/{}".format(settings.IAM_ARN_BASE, _policy_name(bucket_name, readwrite))
+
+
+def bucket_arn(bucket_name):
+    """Return bucket arn e.g. arn:aws:s3:::bucketname"""
+    return "arn:aws:s3:::{}".format(bucket_name)
+
+
+def app_slug(name):
+    """Create a valid slug for s3 using standard django slugify but we add some custom"""
+    return re.sub(r'_+', '-', slugify(name))
+
+
+def get_policy_document(bucket_name, readwrite):
```

Comment:
This is internal only at the moment, prefix with `_`?

Reply:
Yep this makes sense as is real internal utility function.  However some of your other _ prefix recommendations I think are being a bit too strict and confusing a public function which is just not used with internal private functions.